### PR TITLE
Various bugfixes and features

### LINKED ISSUE #4: [버그 🐛]  Makefile clean command need fix
### 오류 종류

컴파일러/메이크파일 오류

### 한 줄 요약

make clean -> does not clean object files in mlx library.

### 발생 과정

```Markdown
1. type make in project's root directory, which will run makefile.
2. type make clean or fclean so that remove object files.
3. type make once again, mlx library doesn't do anything !
4. Because object files in mlx library wasn't removed.
```

### 관련 코드

```C
clean :
    make -C $(LIBMLX_DIRECTORY) clean
    $(RM) $(OBJS_DIRECTORY)

It will be the solution of this problem.
```

### 오류 상태

원인 발견함, 해결 방법 있음

### LINKED ISSUE #28: [Feature ⭐] do not render empty spaces on minimap
### Feature type

Math/graphical feature

### Implementing process / Goal

```Markdown
currently map (int **) has only 2 values: 0 or 1

add 2 or smthn so minimap can see it's empty and ignore
```

### Related code

_No response_

### Note

_No response_

### LINKED ISSUE #30: [Feature ⭐] No seeing-through when game starts !
### Feature type

Engine/event/game mechanics

### Implementing process / Goal

```Markdown
# Problem
When player starts with having position which sticks to the wall, player can see through behind the wall.

# Solution
I simply added player position +0.5 for each coordinate.
(x + 0.5, y + 0.5)

Now playable character is created center of cubic, never see through awkward world behind wall.

Even further, player now can't see outside of map even if it moves straight forward right after game started !

# NOTICE
I tested several maps it works fine, unknown bug may be hidden I couldn't catch.
```

### Related code

```C
static void	world__init(t_world *this, t_string_arr raw_map, t_sizevec map_size)
{
	t_i64vec	it;

	this->camera = (t_camera){
		.pos = {UNSET, UNSET},
		.dir = {0, -1},
		.plane = {0.66, 0},
	};
	this->world_height = map_size.height;
	this->world_width = map_size.width;
	this->worldmap = std__allocate(map_size.height, sizeof(int *));
	it.y = -1;
	while (++it.y < map_size.height)
	{
		it.x = -1;
		this->worldmap[it.y] = std__allocate(map_size.width, sizeof(int));
		while (++it.x < map_size.width)
		{
			this->worldmap[it.y][it.x]
				= (raw_map[it.y][it.x] == MAPFMT__WALL);
			if (mapformat__is_player(raw_map[it.y][it.x]))
				this->camera.pos = (t_vec){it.x + 0.5, it.y + 0.5};
			// TODO: setup func for player position
		}
	}
}
```

### Note

Need to modify existing codes

### PR DESCRIPTION
**Pull request by scarf005**
# Abstract

image

- Fixes #4 
- Resolves #28 
- Resolves #30 

minor bugfixes related to texture and rgb parsing, with some features added

## Types of edition

- [x] Features (not modifying existing features)

# 테스트 결과

- [x] Compile succeed
- [x] Envr diff with 42-cluster
- [x] passed [cub3D map tester](https://github.com/humblEgo/cub3D_map_tester)
- [x] passed norminette

### CODE REVIEW

File: config/packages.mk [hunk 1]
```diff
@@ -17,4 +20,4 @@ endef
 # Sources & Objects
 SRC = $(wildcard src/**/*.c) src/main.c
```
Comment (stripy1026) [inline config/packages.mk]: Using wildcard may be failed on defence !

File: makegen.py [hunk 2]
```diff
@@ -0,0 +1,69 @@
...
+    sections = PKG.read_text().splitlines()
+    begin = 1 + regex_index(sections, r"#\s*@packages", re.IGNORECASE)
+    end = begin + regex_index(sections[begin:], "#")
+    # replace section
+    sections[begin:end] = [template]
+    PKG.write_text("\n".join(sections))
+
+
+if __name__ == "__main__":
+    main()
```
Comment (stripy1026) [inline makegen.py:69]: What is this file exists for ?

**Review by stripy1026 — COMMENTED** [review]
LGTM ! Did you left wildcard in makefile intentionally ? if so, inform me why.

File: makegen.py (hunk 2 above)
Comment (scarf005) [reply makegen.py:69]: as the name says, it writes dependencies (i.e `PKGS`, `rendererV`...) into `makefile`.

**Review by stripy1026 — COMMENTED** [review]
LGTM !

**Pull request merged**